src/api/services/BidActionService.ts

```diff
@@ -752,8 +754,14 @@ export class BidActionService {
             }
             const sellerProfile = sellerProfileModel.toJSON();
 
+            if (!BidRejectReason[reason]) {
+                this.log.error('Unknown BidRejectReason = ${reason}.');
+                throw new MessageException(`Unknown BidRejectReason = ${reason}.`);
+            }
+
             // create the bid reject message
-            const bidMessage = await this.bidFactory.getMessage(BidMessageType.MPA_REJECT, listingItem.hash);
+            const bidMessage = await this.bidFactory.getMessage(BidMessageType.MPA_REJECT, listingItem.hash,
+                [ { id: 'reason', value: reason } ] as IdValuePair[] );
```

Comment:
pass reason: string, not some IdValuePair[], its the factorys responsibility to hide the creation of that.